### PR TITLE
Feature/article canonical urls

### PR DESCRIPTION
**Pull request by nehasri89**
This PR allows Author Profile and Topics articles to have env agnostic URLs. Also RA will have env agnostic URLs too.

There are issues with Context API and enzyme shallow rendering https://github.com/airbnb/enzyme/issues/1647
https://github.com/airbnb/enzyme/pull/1513

So therefore we need to rethink about tests that are failing in Related Articles where we use Context API and Shallow render.

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/1cde2d39aa29f31a7b53323d89008df63127259b/index.html

File: packages/article-list/__tests__/web/article-list-lazy.web.test.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 import {
   addSerializers,
   enzymeRenderedSerializer,
   minimalise
 } from "@times-components/jest-serializer";
+import Context from "@times-components/context";
 import articleListFixture from "../../fixtures/articles.json";
 import adConfig from "../../fixtures/article-ad-config.json";
 import ArticleList from "../../src/article-list";
+import { makeUrl } from "./../utils";
```
Comment (tuncaulubilge) [inline packages/article-list/__tests__/web/article-list-lazy.web.test.js]: could be just "../utils"

File: packages/article-list/__tests__/shared-states.web.js [hunk 2]
```diff
@@ -40,12 +41,14 @@ export default () => {
       test() {
         const apolloError = new ApolloError("Test");
         const testInstance = TestRenderer.create(
-          <ArticleList
-            adConfig={adConfig}
-            emptyStateMessage="Empty State"
-            error={apolloError}
-            refetch={() => {}}
-          />
+          <Context.Provider value={{ makeUrl }}>
```
Comment (tuncaulubilge) [inline packages/article-list/__tests__/shared-states.web.js]: Why don't we add this as a default value instead?

File: packages/author-profile/author-profile.showcase.js [hunk 3]
```diff
@@ -15,6 +16,9 @@ import {
 import storybookReporter from "@times-components/tealium-utils";
 import AuthorProfile from "./src/author-profile";
 
+const makeUrl = ({ slug, shortIdentifier }) =>
+  `https://www.thetimes.co.uk/article/${slug}-${shortIdentifier}`;
```
Comment (tuncaulubilge) [inline packages/author-profile/author-profile.showcase.js]: I'd prefer a default value in context package so we can dry these up

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/de3b2f9b584f308da52d3d3b1b7bd310cb51746b/index.html

File: packages/article-list/__tests__/shared-states.web.js (hunk 2 above)
Comment (nehasri89) [reply packages/article-list/__tests__/shared-states.web.js]: The reason I did not put it in the default is that then it becomes the `default` way to do it, whereas a noop is just saying this is just a default and client's responsible for providing it

File: packages/author-profile/author-profile.showcase.js (hunk 3 above)
Comment (colincclark) [reply packages/author-profile/author-profile.showcase.js]: at least move the function to the `utils` package

File: packages/related-articles/related-articles.showcase.js [hunk 4]
```diff
@@ -17,6 +18,9 @@ import opinionAndTwo1RelatedArticleFixture from "./fixtures/opinionandtwo/1-arti
 import opinionAndTwo2RelatedArticlesFixture from "./fixtures/opinionandtwo/2-articles.js";
 import opinionAndTwo3RelatedArticlesFixture from "./fixtures/opinionandtwo/3-articles.js";
 
+const makeUrl = ({ slug, shortIdentifier }) =>
+  `https://www.thetimes.co.uk/article/${slug}-${shortIdentifier}`;
```
Comment (colincclark) [inline packages/related-articles/related-articles.showcase.js]: move to `utils ` package

File: packages/related-articles/src/related-article-item.web.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
+import React from "react";
```
Comment (colincclark) [inline packages/related-articles/src/related-article-item.web.js]: this whole file is just a duplicate of `related-article-item.js`. Please create a `related-article-item-base.js` file and compose what you need out of that to keep the code DRY

File: packages/topic/topic.showcase.js [hunk 6]
```diff
@@ -6,11 +6,15 @@ import {
...
   topic as makeParams
 } from "@times-components/provider-test-tools";
+import Context from "@times-components/context";
 import storybookReporter from "@times-components/tealium-utils";
 import Topic from "./src/topic";
 import TopicProvider from "../provider/src/topic";
 import adConfig from "./fixtures/topic-ad-config.json";
 
+const makeUrl = ({ slug, shortIdentifier }) =>
+  `https://www.thetimes.co.uk/article/${slug}-${shortIdentifier}`;
```
Comment (colincclark) [inline packages/topic/topic.showcase.js]: `utils` package

File: packages/related-articles/src/related-article-item.web.js (hunk 5 above)
Comment (nehasri89) [reply packages/related-articles/src/related-article-item.web.js]: The plan is to do the same change for the native side pretty soon, so the web file lives only until then. Therefore I would be tempted to make minimal changes for now

Comment (colincclark) [reply packages/related-articles/src/related-article-item.web.js]: I disagree. we either maintain device parity, or keep the codebase DRY and maintainable. Ideally both? @craigbilner to weigh-in?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/31f4ca618a2807c9aa7038bdf805406296066575/index.html

**Comment by craigbilner** [comment]
WIP

**Pull request closed without merging**